sessions/01-exploration-tools/README.md

```diff
@@ -0,0 +1,714 @@
+
+What makes all of these programs work is the Linux dynamic linker/loader. This is a statically linked helper program that resolves symbol names from shared objects at runtime. We can use the dynamic linker to gather information about an executable.
+
+The first and most common thing to do is see what libraries the executable loads, with the `ldd` utility:
+
+```
+$ ldd /bin/ls
+	linux-vdso.so.1 (0x00007ffff13fe000)
+	librt.so.1 => /lib64/librt.so.1 (0x00007fc9b4893000)
+	libacl.so.1 => /lib64/libacl.so.1 (0x00007fc9b468a000)
+	libc.so.6 => /lib64/libc.so.6 (0x00007fc9b42da000)
+	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fc9b40bd000)
+	libattr.so.1 => /lib64/libattr.so.1 (0x00007fc9b3eb8000)
+	/lib64/ld-linux-x86-64.so.2 (0x00007fc9b4a9b000)
+```
+We see that for each dependency in the executable, `ldd` lists where it is found on the filesystem and where it is loaded in the process memory space. Alternatively, you can achieve the same result with the `LD_TRACE_LOADED_OBJECTS` environment variable, or with the dynamic loader itself:
+
+```
+$ LD_TRACE_LOADED_OBJECTS=whatever /bin/ls
+	linux-vdso.so.1 (0x00007fff325fe000)
+	librt.so.1 => /lib64/librt.so.1 (0x00007f1845386000)
+	libacl.so.1 => /lib64/libacl.so.1 (0x00007f184517d000)
+	libc.so.6 => /lib64/libc.so.6 (0x00007f1844dcd000)
+	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f1844bb0000)
+	libattr.so.1 => /lib64/libattr.so.1 (0x00007f18449ab000)
+	/lib64/ld-linux-x86-64.so.2 (0x00007f184558e000)
+$ /lib/ld-linux-x86-64.so.2 --list /bin/ls
+	linux-vdso.so.1 (0x00007fff1e712000)
+	librt.so.1 => /lib64/librt.so.1 (0x00007f18a07d8000)
+	libacl.so.1 => /lib64/libacl.so.1 (0x00007f18a05cf000)
+	libc.so.6 => /lib64/libc.so.6 (0x00007f18a021e000)
+	libattr.so.1 => /lib64/libattr.so.1 (0x00007f189fdfc000)
+	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f18a0001000)
+	/lib64/ld-linux-x86-64.so.2 => /lib/ld-linux-x86-64.so.2 (0x00007f18a0c44000)
+```
+>When using the loader directly, make sure the loader and the executable are compiled for the same platform (e.g. they are both 64-bit or 32-bit).
+
+>You may find out more information about dynamic linker/loader variables in its man page. Issue the command
+`man ld-linux.so`
+and search for the LD_ string to find variables information.
+
+`ldd `shows us **which** libraries are loaded, but it's not any clearer how the loader knows **where** to load them from. First of all, the loader checks every dependency for a slash character. If it finds such a dependency it loads the library from that path, whether it is a relative of absolute path. But it is not the case in our example. For dependencies without slashes, the search order is as follows:
+
+*` DT_RPATH` attribute in the `.dynamic` section of the executable, provided there is no `DT_RUNPATH`; this is deprecated
```

Comment:
Add a space after the `*`. In the current form, this line is not part of the unordered list.